### PR TITLE
Add uncrustify 0.69.0

### PR DESCRIPTION
**Pull request by seanyen**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by seanyen** [comment]
cc @conda-forge/staged-recipes This one is ready for review and merge. Thanks!

File: recipes/uncrustify/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  skip: true  # [osx]
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}              # [not win]
+    - {{ compiler('c') }}                # [not win]
+    - vs2017_win-64                      # [win64]
+    - vs2017_win-32                      # [win32]
```
Comment (beckermr) [inline recipes/uncrustify/meta.yaml]: Someone from @conda-forge/staged-recipes with knowledge of windows will ned to verify these lines.

File: recipes/uncrustify/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/uncrustify/uncrustify/archive/{{ name }}-{{ version }}.tar.gz
+    sha256: 33bd97a07f7c4bd114874f73171aca220bf05c17108f8505a117b97374a347b6
+
+build:
+  number: 0
+  skip: true  # [osx]
```
Comment (beckermr) [inline recipes/uncrustify/meta.yaml]: Can you try removing this skip? I bet the osx library will build fine given what the install script looks like.

File: recipes/uncrustify/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipes/uncrustify/meta.yaml]: If we don't get a comment by the end of today, we can merge.

**Pull request merged**